### DISCUSSION TITLE
Access to requested resource is denied.', 'code': 'Unauthorized'

### DISCUSSION THREAD

**Discussion by eternalparquet** [saleweaver/python-amazon-sp-api#345] [General]
Hello, when I try to access the api i get the error in the title.
I followed the steps here : https://github.com/amzn/selling-partner-api-docs/blob/main/guides/en-US/developer-guide/SellingPartnerApiDeveloperGuide.md
I create the credentials.yml file.
How to solve?

**Comment by eternalparquet** [comment]
Solved, for those who have this problem, check the marketplace arg.

**Reply by Foster2** [reply]
Hi @eternalparquet , did you access to seller central or vendor central?

**Reply by eternalparquet** [reply]
seller central

**Reply by Foster2** [reply]
OK. I have issues to get access to vendor central.  I have followed the steps (for private vendor applications). Yet, the result is just 'Access to requested resource is denied.' I saw that those parameters 'MarketplaceId', 'MarketplaceIds', 'marketplace_ids', 'marketplaceIds' were included in the invocation.

**Reply by eternalparquet** [reply]
> OK. I have issues to get access to vendor central. I have followed the steps (for private vendor applications). Yet, the result is just 'Access to requested resource is denied.' I saw that those parameters 'MarketplaceId', 'MarketplaceIds', 'marketplace_ids', 'marketplaceIds' were included in the invocation.

Please post the invocation code.

**Reply by Foster2** [reply]
Thanks @eternalparquet , I figured it out that vendor central got vendor orders. So, the codes fetching orders should be like ``` res = VendorOrders(credentials=credentials).get_purchase_orders(CreatedAfter=(datetime.utcnow() - timedelta(days=7)).isoformat())```
